src/crimson/os/seastore/cache.cc

```diff
@@ -28,6 +28,24 @@ Cache::~Cache()
   ceph_assert(extents.empty());
 }
 
+Cache::retire_extent_ret Cache::retire_extent_if_cached(
```

Comment:
if the only caller of `Cache::retire_extent_if_cached()` is the ones who get  extents from the cache, the extents being retired should be `CachedExtent`s, so they should be always included by `extents`. i think.

Reply:
No, see above.  dec_ref can be called on addresses that the caller does not have a reference to in memory -- in such cases we just need to make sure that the extent isn't in cache coincidentally.

Reply:
Note that there are two kinds of refcounting going on here:
1) CachedExtents have a refcount to track current memory reference to prevent eviction from the cache.  CachedExtentRef is an alias for intrusive_ptr<CachedExtent> and is generally the mechanism we use to indicate ref count ownership.
2) LBAManager implementations maintain a count of on-disk references to any logical extent.  The TransactionManager interface user uses inc_ref and dec_ref calls to control that refcount.  Right now, that refcount will generally only be 1 or 0, but when we implement clone, it'll track basically the number of rados level objects referencing an extent.

Reply:
> This pathway happens when you need to dec_ref an extent you don't have a memory reference to. For instance, if a sequence of extents are overwritten

if i understand you correctly, we could have a use case, where user rewrites a partial object with a range of, for instance, `[2 * 4KB, 12 * 4KB)`. so we need to decrement these 10 extents' refcounts by 1. as after this change, LBAManager reports

- 5 extents' refcounts are still greater than 0. probably because they were previously cloned.
- 5 other extents' refcounts are now 0 after this change. in which
  - 3 extents were previous read and they are not evicted yet. so we need to retire them after this change. and
  - the other 2 extents were never read. so we don't care about them as they are not in cache. in other words, they are already retired.

Reply:
Yeah, that's the idea.